transport/internet/splithttp/connection.go

```diff
@@ -1,11 +1,112 @@
 package splithttp
 
 import (
+	"bytes"
+	"errors"
+	"fmt"
 	"io"
 	"net"
+	"strconv"
+	"strings"
 	"time"
 )
 
+var (
+	ErrBadRespCode = errors.New("bad response code")
+	BadCodes       = map[int]struct{}{502: {}, 503: {}, 505: {}}
+)
+
+// Optimised to read only response codes
+// Reads response codes until getting EOF or error
+func ConnHttpReadRespCodes(conn net.Conn) (codes []int, err error) {
```

Comment:
on second look, i think this will actually hang if the connection has not yet produced new responses, which is bad since it tries to read infinite amount of responses. I suggest to read one http response with `net/http` everytime there is a new request, and that's it.